libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -107,64 +190,108 @@ bool AP_DDS_Client::create()
+
+        // Data Writer
+        const char* data_writer_ref = topics[i].dw_profile_label;
+        const auto dwriter_req_id = uxr_buffer_create_datawriter_ref(&session,reliable_out,topics[i].dw_id,pub_id,data_writer_ref,UXR_REPLACE);
+
+        // Status requests
+        constexpr uint8_t nRequests = 3;
+        const uint16_t requests[nRequests] = {topic_req_id, pub_req_id, dwriter_req_id};
+        constexpr int requestTimeoutMs = nRequests * maxTimeMsPerRequestMs;
+        uint8_t status[nRequests];
+        if (!uxr_run_session_until_all_status(&session, requestTimeoutMs, requests, status, nRequests)) {
+            GCS_SEND_TEXT(MAV_SEVERITY_ERROR,"XRCE Client: Topic/Pub/Writer session request failure for index 'TODO'");
+            for (int s = 0 ; s < nRequests; s++) {
+                GCS_SEND_TEXT(MAV_SEVERITY_ERROR,"XRCE Client: Status '%d' result '%u'", s, status[s]);
+            }
+            // TODO add a failure log message sharing the status results
+            return false;
+        } else {
+            GCS_SEND_TEXT(MAV_SEVERITY_INFO,"XRCE Client: Topic/Pub/Writer session pass for index 'TOOO'");
+        }
+    }
+
     return true;
 }
 
-void AP_DDS_Client::write()
+void AP_DDS_Client::write_time_topic()
 {
     WITH_SEMAPHORE(csem);
     if (connected) {
         ucdrBuffer ub;
-        uint32_t topic_size = builtin_interfaces_msg_Time_size_of_topic(&time_topic, 0);
-        uxr_prepare_output_stream(&session,reliable_out,dwriter_id,&ub,topic_size);
+        const uint32_t topic_size = builtin_interfaces_msg_Time_size_of_topic(&time_topic, 0);
+        uxr_prepare_output_stream(&session,reliable_out,topics[0].dw_id,&ub,topic_size);
         const bool success = builtin_interfaces_msg_Time_serialize_topic(&ub, &time_topic);
         if (!success) {
             // TODO sometimes serialization fails on bootup. Determine why.
-            // AP_HAL::panic("FATAL: DDS_Client failed to serialize\n");
+            // AP_HAL::panic("FATAL: XRCE_Client failed to serialize\n");
         }
-
     }
+}
 
+void AP_DDS_Client::write_nav_sat_fix_topic()
+{
+    WITH_SEMAPHORE(csem);
+    if (connected) {
+        ucdrBuffer ub;
+        const uint32_t topic_size = sensor_msgs_msg_NavSatFix_size_of_topic(&nav_sat_fix_topic, 0);
+        uxr_prepare_output_stream(&session,reliable_out,topics[1].dw_id,&ub,topic_size);
+        const bool success = sensor_msgs_msg_NavSatFix_serialize_topic(&ub, &nav_sat_fix_topic);
+        if (!success) {
+            // TODO sometimes serialization fails on bootup. Determine why.
+            // AP_HAL::panic("FATAL: DDS_Client failed to serialize\n");
+        }
+    }
 }
 
 void AP_DDS_Client::update()
 {
     WITH_SEMAPHORE(csem);
 
-    update_topic(time_topic);
-    write();
+    const auto cur_time_ms = AP_HAL::millis64();
+
+    if (cur_time_ms - last_time_time_ms > DELAY_TIME_TOPIC_MS) {
```

Comment:
would be nicer to have an update() method on topics table, and have the timing meta-data in the table, would scale a lot better as we then just loop over topics calling update
not essential for this PR if you'd prefer to leave it to later

Reply:
Yea, ironically I actually desire NavSatFix to be at the same rate as the GPS unit itself provides fixes, and with the timestamp of the fix, not when we send. This would apply to most sensor data. 

Additionally, for XRCE DDS Subscribers, there is no rates, it's async based on when companion computer publishes. 

Thus, given the amount of caveats, I'd prefer to leave it out till this can be more generically handled.